tests/scripts/generate_psa_tests.py

```diff
@@ -113,22 +105,26 @@ def read_implemented_dependencies(filename: str) -> FrozenSet[str]:
                      for line in open(filename)
                      for symbol in re.findall(r'\bPSA_WANT_\w+\b', line))
 _implemented_dependencies = None #type: Optional[FrozenSet[str]] #pylint: disable=invalid-name
-# This is a temporary fix for the KEY_PAIR_LEGACY symbols since they are not
-# defined in "crypto_config.h". This fix can be removed as soon as these _LEGACY
-# symbols will be removed from the code.
-_LEGACY_KEY_PAIR = ['MBEDTLS_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_LEGACY',
-                    'MBEDTLS_PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_LEGACY']
 def hack_dependencies_not_implemented(dependencies: List[str]) -> None:
     global _implemented_dependencies #pylint: disable=global-statement,invalid-name
     if _implemented_dependencies is None:
         _implemented_dependencies = \
             read_implemented_dependencies('include/psa/crypto_config.h')
     if not all((dep.lstrip('!') in _implemented_dependencies or
-                'PSA_WANT' not in dep or
-                dep.lstrip('!') in _LEGACY_KEY_PAIR)
+                'PSA_WANT' not in dep)
                for dep in dependencies):
         dependencies.append('DEPENDENCY_NOT_IMPLEMENTED_YET')
 
+# PSA_WANT_KEY_TYPE_xxx_KEY_PAIR symbols have a GENERATE suffix to state that
+# they support key generation.
+def fix_key_pair_dependencies(dep_list: List[str], usage: str):
```

Comment:
The code to turn PSA mechanism identifiers (`PSA_ALG_xxx`, `PSA_KEY_TYPE_xxx`, etc.) started out simple, but over the time it's grown more complex: ECC curves, now key pairs split into usage. And in parallel we're likely to want to use that in places other than the systematic operation tests here (e.g. #7803 and follow-ups). There's some similar code bit-rotting in `tests/scripts/set_psa_test_dependencies.py`, which we ran locally to convert some legacy dependencies into PSA dependencies but don't use on a regular basis. We should make a unified interface.

I think this counts as partly on the time budget for driver-only ECC and partly on the time budget for PBKDF2, since those are the two projects that need the feature now. I can help with the Python code.

Reply:
@gilles-peskine-arm I'm not so familiar with those scripts. Do you think you could define an issue with scope that's sensible and fair to add to one of the driver-only ECC EPICs (and similarly for PBKDF2)? Also, should this wait for #7812 or is it independent enough?

Reply:
I've filed an issue https://github.com/Mbed-TLS/mbedtls/issues/7845 but I have a hard time giving it a proper scope because I don't know what the interface should look like. It's a case where I'll know it if I do it. So more of a time-boxed task, but one that'll require less time if I do it because I wrote most of the existing code.

Reply:
I don't think it'll make any difference to #7812. It might mean fewer places where #7812 will need to inject special handling, but that special handling would already exist for RSA and ECC so it would be very easy to find.